src/nirfsg/system_tests/test_system_nirfsg.py

```diff
@@ -0,0 +1,559 @@
+        test_rfsg_device_session.export_signal(nirfsg.Signal.DONE_EVENT, '', 'PFI0')
+        assert test_rfsg_device_session.exported_done_event_output_terminal == 'PFI0'
+
+    def test_export_signal_with_invalid_signal(self, test_rfsg_device_session):
+        try:
+            test_rfsg_device_session.export_signal(nirfsg.Signal.INVALID, '', 'PXI_Trig0')
+            assert False
+        except AttributeError:
+            pass
+
+    @pytest.mark.skipif(use_simulated_session is True, reason="RoCo is not invoked for simulated device")
+    def test_export_signal_with_invalid_terminal(self, test_rfsg_device_session):
+        try:
+            test_rfsg_device_session.export_signal(nirfsg.Signal.START_TRIGGER, '', 'InvalidTerminal')
+            test_rfsg_device_session.commit()
+            assert False
+        except nirfsg.Error as e:
+            assert e.code == -1074107490
+            assert 'Destination terminal to be routed could not be found on the device' in e.description
+
+    def test_save_load_configuration(self, test_rfsg_device_session):
+        test_rfsg_device_session.configure_rf(2e9, -5.0)
+        test_rfsg_device_session.iq_rate = 1e6
+        test_rfsg_device_session.save_configurations_to_file(get_test_file_path('tempConfiguration.json'))
+        assert os.path.exists(get_test_file_path('tempConfiguration.json'))
+        test_rfsg_device_session.configure_rf(3e9, -15.0)
+        test_rfsg_device_session.iq_rate = 2e6
+        assert test_rfsg_device_session.frequency == 3e9
+        assert test_rfsg_device_session.power_level == -15.0
+        assert test_rfsg_device_session.iq_rate == 2e6
+        test_rfsg_device_session.load_configurations_from_file(get_test_file_path('tempConfiguration.json'))
+        assert test_rfsg_device_session.frequency == 2e9
+        assert test_rfsg_device_session.power_level == -5.0
+        assert test_rfsg_device_session.iq_rate == 1e6
+        os.remove(get_test_file_path('tempConfiguration.json'))
+
+# Basic tests for generation
+    @pytest.mark.skipif(use_simulated_session is False, reason="Test executed with status check in real hw")
+    def test_cw_generation(self, test_rfsg_device_session):
+        test_rfsg_device_session.configure_rf(2e9, -5.0)
+        with test_rfsg_device_session.initiate():
+            test_rfsg_device_session.check_generation_status()
+
+    @pytest.mark.skipif(use_simulated_session is True, reason="isdone is always True on simulated device")
+    def test_cw_generation_with_status(self, test_rfsg_device_session):
+        test_rfsg_device_session.configure_rf(2e9, -5.0)
+        with test_rfsg_device_session.initiate():
+            isdone = test_rfsg_device_session.check_generation_status()
+            assert isdone is False  # isdone will never be True in CW mode
+
+    @pytest.mark.skipif(use_simulated_session is True, reason="isdone is always True on simulated device")
```

Comment:
>isdone is always True on simulated device

That shouldn't stop you from calling abort to make sure nothing bad happens.
Which is all you really need to test. Remember: this is an API test, not a behavioral test.
If simulation changes something when you call, great. Do your assert.
If not, don't sweat it. Call the function, anyway.

By the way, I consider this more of a test of the context manager.
The nifake unit test `test_acquisition_context_manager` has that covered.
You should avoid the context manager for your test and call abort explicitly, instead.

I'll admit, though, most of our APIs don't have a test_abort test, for whatever reason.